cyhy_commander/nmap/nmap_importer.py

```diff
@@ -109,50 +112,93 @@ def __store_port_details(self, parsed_host):
                 self.__ticket_manager.port_open(ip, port)
-            details["ip"] = ip
-            details["ip_int"] = long(ip)
             details["port"] = port
-            details["time"] = time
+            details["time"] = parsed_host["endtime"]
             details["source"] = NmapImporter.SOURCE
-            details["owner"] = ip_owner
             details["latest"] = True
-            report = self.__db.PortScanDoc()
-            util.copy_attrs(details, report)
-            report.save()
-            if details.get("service", {}).get("name") in RISKY_SERVICES:
-                report["source_id"] = RISKY_SERVICES_SOURCE_ID
-                report["name"] = "Potentially Risky Service Detected: {}".format(
-                    details["service"]["name"]
-                )
-                report["service"] = details["service"]["name"]
-                self.__ticket_manager.open_ticket(
-                    report, "potentially risky service detected"
-                )
+
+            if host_doc and host_doc.get("hostnames"):
+                # Create a PortScanDoc for each hostname/owner combination
+                for h in host_doc["hostnames"]:
+                    report = self.__db.PortScanDoc()
+                    util.copy_attrs(details, report)
+                    report["owner"] = h.get("owner", UNKNOWN_OWNER)
+                    report["hostname"] = h["hostname"]
+                    report.ip = ip  # sets ip and ip_int
+                    report.save()
+                    if details.get("service", {}).get("name") in RISKY_SERVICES:
+                        report["source_id"] = RISKY_SERVICES_SOURCE_ID
+                        report["name"] = "Potentially Risky Service Detected: {}".format(
+                            details["service"]["name"]
+                        )
+                        report["service"] = details["service"]["name"]
+                        # Open a ticket for this hostname/owner combination
+                        self.__ticket_manager.open_ticket(
+                            report, "potentially risky service detected"
+                        )
+            else:
+                # There are no hostnames in the HostDoc, so create a single
+                # PortScanDoc with no hostname that is owned by the IP owner
+                report = self.__db.PortScanDoc()
+                util.copy_attrs(details, report)
+                report["owner"] = ip_owner
+                report["hostname"] = None
+                report.ip = ip  # sets ip and ip_int
+                report.save()
+                if details.get("service", {}).get("name") in RISKY_SERVICES:
+                    # If IP owner is known and is not the default owner, open a
+                    # ticket
+                    if (
+                        ip_owner != DEFAULT_OWNER
+                        and ip_owner != UNKNOWN_OWNER
+                        and ip_owner is not None
```

Comment:
I personally believe a check for `None` should always come first since it precludes anything else. The `not in` is just because it's two `!=` checks in an `and` chain.

```suggestion
                        ip_owner is not None
                        and ip_owner not in [DEFAULT_OWNER, UNKNOWN_OWNER]
```

Reply:
Agreed- your way is an improvement.